en/docs/concepts/oidc/hybrid-client-profile.md

```diff
@@ -0,0 +1,242 @@
+```
+
+When you decrypt the ` id_token ` it will be as follows:
+
+``` java
+{
+ “at_hash”: “JrZY9MtYVEIIJUx-DDBmww”,
+ “c_hash”: “S5UOXRNNyYsI6Z0G3xxdpw”,
+ “sub”: “admin”,
+ “aud”: [
+ “nczbg5m5xxt6tP4UMZwB6PtQoQoa”
+ ],
+ “azp”: “nczbg5m5xxt6tP4UMZwB6PtQoQoa”,
+ “iss”: “https://localhost:9443/oauth2/token",
+ “exp”: 1510831512,
+ “nonce”: “asd”,
+ “iat”: 1510831508
+}
+```
+
+!!! tip 
+    <p>Here, the id\_token is required to have a c\_hash value.</p>
+ 
+    <p>c\_hash is the base64url encoding of the left-most half of a hash of the
+    octets in the ASCII representation of a code value, where the hash
+    algorithm used is the hash algorithm of the ` alg `
+    header parameter of the ID token’s JOSE header.</p>
+ 
+    <p>The c\_hash value is mandatory when an ` id_token ` is
+    issued with code, and the ` response_type ` is equal to
+    ` code id_token ` or
+    ` code id_token token ` .</p>
+ 
+    <p>You can send the code to the token endpoint to request for an access
+    token, refresh token and id\_token. For this you can use the same curl
+    command provided for the code token specified
+    [above](#curl)."</p>
+ 
+
+Following is the response that you will receive from token endpoint:
+
+``` java
+{“access_token”:”1940a308-d492–3660-a9f8–46723cc582e9",”refresh_token”:”6b96cc3a-00da-3d7d-acd1–5aaf76dcd9d4",”scope”:”openid”,”id_token”:”eyJ4NXQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJraWQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJhbGciOiJSUzI1NiJ9.eyJhdF9oYXNoIjoiSnJaWTlNdFlWRUlJSlV4LUREQm13dyIsInN1YiI6ImFkbWluIiwiYXVkIjpbIm5jemJnNW01eHh0NnRQNFVNWndCNlB0UW9Rb2EiXSwiYXpwIjoibmN6Ymc1bTV4eHQ2dFA0VU1ad0I2UHRRb1FvYSIsImlzcyI6Imh0dHBzOlwvXC9sb2NhbGhvc3Q6OTQ0M1wvb2F1dGgyXC90b2tlbiIsImV4cCI6MTUxMDgzMjI3MSwibm9uY2UiOiJhc2QiLCJpYXQiOjE1MTA4MzIyNjd9.jAGLp8FFdIyFi4ZmvRPX9hVu8NbLVL2iM1895UNrS7wqgl2PCi7zHnvBoOYkbsxxMYGoVepFNzz7hHbk-kuzq_kBoBsZK2Ucbv0hUkwiEkigLy6hpGm-mqXjai3cjlJevWOVcZbMhkEyRlsZtdUG0RCzteT7emAuZLFm5zfMpq1h5JsVRGjK_6fQbHhB2Svkl_kV_ctAD8_kymASGEjRGnwGW5np4uBI0NPYMDTvrl8N9i6yfUVD9-y7rL9Gtrq9hK28Swj5Szvv_c1IX8wYBP-p8gu2cBpGIulIq-OkbfCUh-rrbh96relOaKwKwk0g7nST6o6wZTAwaicNQBYHYw”,”token_type”:”Bearer”,”expires_in”:298234}
+```
+
+When you decrypt the ` id_token ` it will be as follows:
+
+``` java
+{
+ “at_hash”: “JrZY9MtYVEIIJUx-DDBmww”,
+ “sub”: “admin”,
+ “aud”: [
+ “nczbg5m5xxt6tP4UMZwB6PtQoQoa”
+ ],
+ “azp”: “nczbg5m5xxt6tP4UMZwB6PtQoQoa”,
+ “iss”: “https://localhost:9443/oauth2/token",
+ “exp”: 1510832271,
+ “nonce”: “asd”,
+ “iat”: 1510832267
+}
+```
+<a name="validations"></a>
+In case there are two id\_tokens issued, where one id\_token is from
+authorization endpoint and other is from token endpoint, be sure to
+perform the following validations based on the OpenID Connect
+specification:
```

Comment:
In case there are two id\_tokens issued; where one id\_token is from
the authorization endpoint, and other is from the token endpoint, be sure to
perform the following validations based on the OpenID Connect
specification.